pkg/ds/stack.go

```diff
@@ -0,0 +1,35 @@
+package ds
+
+type Stack[T any] struct {
+	keys []T
+}
+
+func NewStack[T any]() *Stack[T] {
+	return &Stack[T]{nil}
```

Comment:
We will have a nil entry in the stack. May lead to NPE. Is this desirable?

Reply:
the zero value of slice is `nil` so it won't affect the slice operations like `append` `len` etc
Ref - https://dev.to/melvin2016/what-is-the-empty-or-zero-value-for-a-slice-in-go-or-golang-4hl4

`&Stack[T]{nil}` and `&Stack[T]{}` will both result in same behavior

Reply:
I've copied it from https://github.com/ttdsuen/golang-stack/blob/main/stack.go btw